### PR TITLE
feat: front lead 2 and 2

### PR DESCRIPTION
**Pull request by c-j-j**
Creating draft PR for visibility/early feedback.

The plan for this PR is to have the lead 2 and 2 front-page working for all breakpoints and on landscape/portrait.

### Oustanding issues (later for a PR)
- [ ] Figure out better solution for how to have correct teaser font size on landscape/portrait and medium/wide/huge - do we need to use the font size logic in styleguide (which doesn't yet support such a thing)
- [ ] Byline has incorrect marginBottom
- [ ] Small breakpoint is broken - but as it isn't going to show a front page I'm not sure if this is an issue?
- [ ] Not enough room to display full headlines on an iPad mini
- [ ] Text generally doesn't align to baseline. RemoveWhitespace seems to be overly cautious with how much space to allocate.

### Orientation/Breakpoint limitation

Currently, we are using breakpoints medium/wide/huge. These are incompatible with the designs we have been given as they have differences for landscape/portrait. For example, Tile-X's headline should have 45px on landscape and 50px on portrait when the `wide` breakpoint is hit... so I've had to assign different font sizes based on orientation as well as breakpoint.

There's also an issue with the existing slices. Lead 2 and 2 no pic - for example, only uses breakpoints to decide which layout to show... which results in the landscape template being shown on an iPad Pro 12.9 in portrait because it hits that breakpoint.

Also, in the styleguide package, there is font sizing logic that uses the font-size passed in from native, however, I don't think this is currently used for article teaser text to show larger fonts when the `medium`/`large`/`xlarge` options are set.

Potential solution: In the Responsive context provider, we also provide a `deviceSize` which can be small/medium/large etc, and then use that value to decide how large text should be. We then will only need to factor in edge cases when landscape/portrait have different font sizes (for e.g. tile E has different headline font size)

### New Components Created/Copied 

**Copied Tile X/Y/D/E/AL to Tile-X-Front/etc** 
The front tiles will have the following differences:
- Always show byline, and will be before the content
- Will not need save star/labels/flags
- Will use the new `FrontTileSummary` component
- Have justified/left-aligned text based on template
- Will eventually have columns
- Subtle styling differences - for e.g. different font sizes for landscape/portrait/
- Uses article content as teaser text, instead of the summary800 fields, as some tiles will need a lot of teaser text.

**Copied TileSummary to FrontTileSummary**
- Uses new `FrontArticleSummaryContent` to render the content
- Uses new `FrontArticleSummary`
- Does not render flags/save star

**New FrontArticleSummaryContent**
This does the following, and wraps the original `ArticleSummaryContent` component
- indents paragraphs
- different font styles for landscape/portrait - which we might want to duplicate in `ArticleSummaryContent`

iPad mini in portrait
image

iPad mini in landscape
image

### CODE REVIEW

File: packages/edition-slices/src/tiles/tile-x-front/styles/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const fontSizePortraitResolver = {
+  [editionBreakpoints.medium]: 40,
+  [editionBreakpoints.wide]: 50,
+  [editionBreakpoints.huge]: 50,
+};
+
+export default (breakpoint) => ({
+  container: {
+    flex: 1,
+    padding: spacing(2),
```
Comment (aronshamash) [inline packages/edition-slices/src/tiles/tile-x-front/styles/index.js:24]: can just be `paddingRight: spacing(2), paddingBottom: spacing(2)` I think.

File: packages/edition-slices/src/tiles/tile-x-front/styles/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import {
+  colours,
+  fonts,
+  spacing,
+  editionBreakpoints,
+  globalSpacingStyles,
+} from "@times-components-native/styleguide";
+
+const fontSizeLandscapeResolver = {
```
Comment (aronshamash) [inline packages/edition-slices/src/tiles/tile-x-front/styles/index.js:9]: what about combining this to 1 object for both landscape and portrait ?

File: packages/front-page/front-article-summary-content/styles.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import styleguide, {
+  editionBreakpoints,
+} from "@times-components-native/styleguide";
+
+const { colours, fonts, spacing } = styleguide();
+
+const fontSizeLandscapeResolver = {
```
Comment (aronshamash) [inline packages/front-page/front-article-summary-content/styles.js:7]: these 4 objects can be combined I think.

File: packages/slice-layout/src/templates/frontleadtwoandtwo/index.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import { Dimensions, View } from "react-native";
+import { editionBreakpoints } from "@times-components-native/styleguide";
+import { ItemColSeparator } from "../shared";
+import stylesFactory from "./styles";
+
+import VerticalLayout from "../verticallayout";
+
+function calculateContentWidth(windowWidth) {
```
Comment (aronshamash) [inline packages/slice-layout/src/templates/frontleadtwoandtwo/index.js:9]: could this be done in an object literal to reduce complexity ?

File: packages/edition-slices/src/tiles/tile-x-front/styles/index.js (hunk 2 above)
Comment (c-j-j) [reply packages/edition-slices/src/tiles/tile-x-front/styles/index.js:9]: Unsure how that'll work, as we access these fields with the breakpoint and they're different for landscape/portrait:

```
fontSizeLandscapeResolver[breakpoint]
```

File: packages/edition-slices/src/tiles/tile-x-front/styles/index.js (hunk 1 above)
Comment (c-j-j) [reply packages/edition-slices/src/tiles/tile-x-front/styles/index.js:24]: Yep that's nicer.

File: packages/slice-layout/src/templates/frontleadtwoandtwo/index.js (hunk 4 above)
Comment (c-j-j) [reply packages/slice-layout/src/templates/frontleadtwoandtwo/index.js:9]: Happy to change this... How would that work with an object? 

I thought this needed to be done via if statements, as the order of the checks is important. For example, we first need to check the widest width and then check the narrower widths.

Comment (aronshamash) [reply packages/slice-layout/src/templates/frontleadtwoandtwo/index.js:9]: hmm yes looking at it closer it makes sense to keep as you have it!

File: packages/edition-slices/src/tiles/tile-x-front/styles/index.js (hunk 2 above)
Comment (aronshamash) [reply packages/edition-slices/src/tiles/tile-x-front/styles/index.js:9]:
```
const fontSizeResolver = {
  [editionBreakpoints.medium]: {
   landscape: 40,
   portrait: 40,
   },
  [editionBreakpoints.wide]: {
   landscape: 40,
   portrait: 50,
   },
  [editionBreakpoints.huge]: {
   landscape: 50,
   portrait: 50,
   }
}
```
than access by `fontSizeResolver[breakpoint].landscape`

Comment (c-j-j) [reply packages/edition-slices/src/tiles/tile-x-front/styles/index.js:9]: Yeah that sounds good - makes it easier to see the differences between landscape and portrait

File: packages/front-page/front-article-summary-content/styles.js (hunk 3 above)
Comment (c-j-j) [reply packages/front-page/front-article-summary-content/styles.js:7]: 2 objects might work better for this, one for font size and one for line height. Similar to https://github.com/newsuk/times-components-native/pull/73#discussion_r466454197 - what do you think?

File: .eslintrc.js [hunk 5]
```diff
@@ -85,6 +85,7 @@ module.exports = {
     "@typescript-eslint/no-inferrable-types": "off",
     "@typescript-eslint/no-non-null-assertion": "off",
     // Typescript specific to revist when majority TS codebase
+    "@typescript-eslint/ban-ts-comment": "off",
```
Comment (Geoff-Ford) [inline .eslintrc.js:88]: Can you bump this above the comment as I can't see us never needing to have some ts comments

File: packages/edition-slices/src/slices/frontleadtwoandtwo/index.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  // const {
+  //   onPress,
+  // slice: { lead1, lead2, support1, support2 },
+  // } = props;
+
+  const onPress = () => {
+    console.log("PRESSED");
+  };
+
+  const { lead1, lead2, support1, support2 } = testSlice;
```
Comment (Geoff-Ford) [inline packages/edition-slices/src/slices/frontleadtwoandtwo/index.js:23]: Is this `onPress`, the above `testSlice` to be removed in favour of the commented code above?

File: packages/edition-slices/src/slices/frontleadtwoandtwo/index.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    />
+  );
+}
+
+const FrontLeadTwoAndTwo = (props) => {
+  return (
+    <ResponsiveSlice
+      renderSmall={(breakpoint, orientation) =>
+        renderMedium(props, breakpoint, orientation)
+      }
```
Comment (Geoff-Ford) [inline packages/edition-slices/src/slices/frontleadtwoandtwo/index.js:84]: Could you maybe add a function above the `return` here like:

```
const renderSlice = (breakpoint, orientation) => renderMedium(props, breakpoint, orientation);
```

Then each of these render props could just take the that function, like:

```
...
renderSmall={renderSlice}
...
```

Not a huge saving granted but saves some repetition as they are all using `renderMedium`

File: packages/edition-slices/src/slices/frontleadtwoandtwo/testSlice.js [hunk 8]
```diff
@@ -0,0 +1,3703 @@
+export const testSlice = {
```
Comment (Geoff-Ford) [inline packages/edition-slices/src/slices/frontleadtwoandtwo/testSlice.js:1]: Is this to go (see above comment)

File: packages/edition-slices/src/tiles/tile-al-front/index.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  onPress,
+  tile,
+  breakpoint = editionBreakpoints.wide,
+}) => {
+  const crop = getTileImage(tile, "crop32");
+  const styles = stylesFactory(breakpoint);
+
+  if (!crop) {
+    return null;
+  }
```
Comment (Geoff-Ford) [inline packages/edition-slices/src/tiles/tile-al-front/index.js:27]: One liner

File: packages/edition-slices/src/tiles/tile-d-front/index.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  onPress,
+  tile,
+  breakpoint = editionBreakpoints.small,
+}) => {
+  const crop = getTileImage(tile, "crop32");
+  const styles = stylesFactory(breakpoint);
+
+  if (!crop) {
+    return null;
+  }
```
Comment (Geoff-Ford) [inline packages/edition-slices/src/tiles/tile-d-front/index.js:26]: Guessing there might be a few of these 😝

File: packages/front-page/front-tile-summary.tsx [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+  containerStyle?: any;
+  summary: Markup;
+  summaryStyle?: any;
+  tile: any;
+  bylines?: Markup;
+  whiteSpaceHeight: number;
+  linesOfTeaserToRender?: number | undefined;
+}
+
+class FrontTileSummary extends Component<Props> {
```
Comment (Geoff-Ford) [inline packages/front-page/front-tile-summary.tsx:27]: 😱  Must this be done as a `class`?

File: packages/front-page/styles/index.js [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    ...textStyle,
+    fontSize: fontSizeLandscapeResolver[breakpoint],
+    lineHeight: lineHeightLandscapeResolver[breakpoint],
+  },
+  textPortrait: {
+    ...textStyle,
+    fontSize: fontSizePortraitResolver[breakpoint],
+    lineHeight: lineHeightPortraitResolver[breakpoint],
+  },
+  bylineContainer: { marginBottom: 5 },
```
Comment (Geoff-Ford) [inline packages/front-page/styles/index.js:47]: Should this be `spacing(1)`?

File: packages/slice-layout/src/templates/frontleadtwoandtwo/index.js [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+  if (windowWidth >= 1112) {
+    return 20;
+  }
+  if (windowWidth >= 1080) {
+    return 15;
+  }
+
+  if (windowWidth >= 1024) {
+    return 15;
+  }
```
Comment (Geoff-Ford) [inline packages/slice-layout/src/templates/frontleadtwoandtwo/index.js:46]: All the above can be one liners no?

File: packages/front-page/front-article-summary-content/styles.js [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+    ...textStyle,
+    fontSize: fontSizeResolver[breakpoint].landscape,
+    lineHeight: lineHeightResolver[breakpoint].landscape,
+  },
+  textPortrait: {
+    ...textStyle,
+    fontSize: fontSizeResolver[breakpoint].portrait,
+    lineHeight: lineHeightResolver[breakpoint].portrait,
+  },
+  bylineContainer: { marginBottom: 5 },
```
Comment (Geoff-Ford) [inline packages/front-page/front-article-summary-content/styles.js:53]: Should use `spacing(1)`

File: packages/edition-slices/src/slices/frontleadtwoandtwo/index.js (hunk 6 above)
Comment (c-j-j) [reply packages/edition-slices/src/slices/frontleadtwoandtwo/index.js:23]: Yes, exactly. This is the only way I could figure out how render the new slice in the app. The tiles don't render well on storybook as they get overfilled with text (hoping to look into that today)

File: packages/front-page/front-tile-summary.tsx (hunk 11 above)
Comment (c-j-j) [reply packages/front-page/front-tile-summary.tsx:27]: I copied/pasted the original TileSummary - I can change to a functional one

File: packages/front-page/styles/index.js (hunk 12 above)
Comment (c-j-j) [reply packages/front-page/styles/index.js:47]: Yes good spot

File: packages/edition-slices/src/slices/frontleadtwoandtwo/testSlice.js (hunk 8 above)
Comment (c-j-j) [reply packages/edition-slices/src/slices/frontleadtwoandtwo/testSlice.js:1]: Yes

File: packages/front-page/front-article-summary-content/styles.js (hunk 3 above)
Comment (aronshamash) [reply packages/front-page/front-article-summary-content/styles.js]: yes sounds better

File: packages/front-page/front-tile-summary.tsx (hunk 11 above)
Comment (c-j-j) [reply packages/front-page/front-tile-summary.tsx]: Done

File: packages/edition-slices/src/tiles/tile-e-front/index.js [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+  showSummary,
+  showByline,
+  breakpoint = editionBreakpoints.small,
+}) => {
+  const crop = getTileImage(tile, "crop45");
+  const styles = stylesFactory(breakpoint);
+
+  if (!crop) {
+    return null;
+  }
```
Comment (Geoff-Ford) [inline packages/edition-slices/src/tiles/tile-e-front/index.js:29]: Thought you might have found the other ones ;-)

**Pull request merged**